game.rb

```diff
@@ -21,13 +21,23 @@ def reset
     @board.new_game
   end
 
+  def chat_select(coordinates)
+    if not board.valid_coordinates?(coordinates)
+      puts "invalid coordinates"
+    end
+    board.reveal_at(coordinates)
+  end
+
```

Comment:
This is where we have to be super careful, as you're modifying game state from a thread. This should be migrated to shuffle the commands onto a queue and pop them off in the update method of the game class.

```ruby
def chat_select(coords)
# @queue = Queue.new in #reset
@queue.push cmd
end

def update
  while cmd = queue.pop
    process_command(cmd)
  end
end
```

something like that